client/post-editor/editor-ground-control/history-button.jsx

```diff
@@ -46,10 +41,18 @@ HistoryButton.propTypes = {
 	translate: PropTypes.func,
 };
 
+const selectHistory = () => dispatch => {
+	dispatch(
+		withAnalytics(
+			recordTracksEvent( 'calypso_editor_history_button_click' ),
+			openPostRevisionsDialog()
+		)
+	);
+};
+
 export default flow(
 	localize,
 	connect( null, {
-		openDialog: openPostRevisionsDialog,
-		recordTracksEvent,
+		selectHistory,
```

Comment:
it may be easier to slough through the logic if we create an explicit `mapDispatchToProps` function. we can probably do away with `flow()` here too since it's not adding anything (actually in this case it's only adding more code)

```js
const mapDispatchToProps = dispatch => ( {
  selectHistory: () => dispatch( withAnalytics(
    recordTracksEvent( 'calypso_editor_history_button_click' ),
    openPostRevisionsDialog()
  ) )
} )

export default connect( null, mapDispatchToProps )( localize( HistoryButton ) );
```

the was it's written in the PR is that `selectHistory` is a function which when called returns a function expecting a dispatch function. in reality, we enclose around the dispatch function inside of `connect()` and instead want to provide a function that when called actually dispatches the actions.